v2.3.4, Add support for other exchange types and options

linus opened the pull request:
- behaviour.exchangeType specifies exchange type, default "topic"
- behaviour.exchangeOptions specifies exchange options

linus committed:
Add support for other exchange types and options

- behaviour.exchangeType specifies exchange type, default "topic"
- behaviour.exchangeOptions specifies exchange options

linus committed:
v2.3.4

norla commented on index.js:
```diff
@@ -14,6 +14,8 @@ var dummyLogger = {
 
 var defaultBehaviour = {
+  exchangeType: "topic",
+  exchangeOptions: {},
```
We have the same functionality for queues, that prop is called "queueArguments". Perhaps we can name this "exchangeArguments" for consistency. ("exchangeOptions" is a better name per se, but I think consistency is more important here)

linus replied:
Thanks, I saw that. Here we actually send in the whole options object though, the reason being that exp-amqp-connection does not use the other options (as opposed to the queue options). I am happy to change this to something like `{ arguments: behaviour.exchangeArguments }` though, if you prefer.

norla replied:
You are correct, the queue thing is a different case. Better to keep "exchangeOptions" then.

norla commented:
Add examples/docs in readme as well, and also some tests.
There is also a line in the readme that says the exchange type is always "topic", remove it plz.

linus committed:
Document exchangeType, exchangeOptions in README

linus committed:
Pass err in shutdown, test for alternate exchange option

linus committed:
Assert publisher exchange with same parameters as subscriber

- Publisher (producer) exchange needs to be asserted with type and
  options just as subscriber (consumer) exchange.

linus committed:
Tests for exchange type and options

- Fix test for exchange with options
- Add test for exchange of different type (fanout)

linus commented:
@norla please have another look at README and tests when convenient.
